documentation/src/main/wiki/ActionKnot.md

```diff
@@ -17,20 +54,97 @@ Example:
 </script>
 ```
 
-ActionKnot processes only those fragments that `<script>` tag has defined `data-api-type="form-{NAME}"` parameter, 
-where `{NAME}` is a unique name of a form (assuming there may be more than one form on a single page it is used to distinguish requested snippet).
-In the example above, `{NAME}` is `1`.
+Now we can explain how and why this additional hidden input `_frmId` with a value `1` appears . It
+is automatically added by Action Knot and is used to distinguish a requested form during submission process 
+(there could be more than one form at the same template). Its value comes from a script's `data-api-type`
+attribute - it retrieve a `{NAME}` value from `data-api-type="form-{NAME}"`.
 
 Following data attributes are available in the `<form>` tag with described purpose:
-
 - `data-knotx-action` - this is a name of an [[Action Adapter|ActionAdapter]] that will be used to handle submitted data. 
-It is similar concept as `data-service-{NAME}` in [[View Knot|ViewKnot]]. In the example above, 
-Action Handler registered under name `step1` will handle this form data submit.
-- `data-knotx-on-{SIGNAL}` - name of a [Signal](#Sginal) that should be applied.
+It is similar concept as `data-service-{NAME}` in [[View Knot|ViewKnot]]. In the example, 
+Action Handler registered under name `step1` will handle this form data submission.
+- `data-knotx-on-{SIGNAL}` - name of a [Signal](#Signal) that should be applied. In the example 
+there is only one signal success with the value `'_self'` which means that after correct response 
+the client will stay on the same page.
 
-#### Signal
+### Signal
 Signal is basically a decision about further request processing. Value of the signal can be either:
-
 - `url` of a page that user should be redirected to after processing form submit,
```

Comment:
Maybe instead of `url` we should use `path` or explain that full url is not required?